airflow/providers/google/suite/hooks/drive.py

```diff
@@ -243,8 +284,13 @@ def upload_file(
             .create(body=file_metadata, media_body=media, fields="id", supportsAllDrives=True)
             .execute(num_retries=self.num_retries)
         )
-        self.log.info("File %s uploaded to gdrive://%s.", local_location, remote_location)
-        return file.get("id")
+        file_id = file.get("id")
+
+        upload_location = (
```

Comment:
I think having to do the traverse up the directories hierarchy is a convenience and I think maybe it should be guarded by a parameter ("show_full_target_path") in the `upload` method which might be set to `true` by default?

If not for the performance, it will also give the escape hatch in case for some reason the traversal will be problematic. I can imagine a number of cases - for example it could be that the file is in a place that you can upload a file but you can only have permissions to look at some part of the tree. Not sure if it might happen in GDrive, but this can - for example - happen in regular POSIX filesystem.

Reply:
Well two questions about this:

1. In a scenario where user only has permission to look at part of the tree, would the API throw an error or would we just not be able to see it at all?

2. Are there people who have such deep traversal trees in their Google Drive that this would be a large amount of time? How common is that?

Reply:
I think it would be worth us investigating whether this is actually a scenario that could happen in Google drive as I have a feeling that the Google drive filesystem is probably closer to an s3 than a POSIX

Reply:
BTW. I have no idea how it works, and  whether it's possible, just raised it as possible scenario where such traversal might fail. There might be other cases (is this possible there is more than one parent maybe? Or some rate limiting when calling an API). So I am not really sure if the scenario is possible, and which scenario might be problematic, my only point here was that we shoulkd  have an escape hatch in such case. This traversal is only needed to log the full destination path, it's not used anywhere else, so IMHO we should be able to disable it in case any of similar things happen. I am not sure if those are likely scenarios/issues currently (they might also change in the future), so without spending a lot of time on analysing those, having a simple to use escape hatch seems to be a good idea :D)

Reply:
@dimberman I have tested your first scenario. Google API doesn't return more `parents` than user can see. So having path: `/Drive/FolderA/FolderB/file.csv` and user is granted access to `FolderB` then what he sees is `/FolderB/file.csv` which makes sense because this is what he sees on the UI as well.

I do agree that for performance reasons it is nice to switch it off when needed in case someone has a large tree.